File: lcp/src/main/java/org/readium/lcp/DRMContext.kt
package org.readium.lcp

data class DRMContext (
    // Hashed passphrase in hex used to generate the user key
    val hashedPassphrase: String,


    // Hex key, encoded with user key, used to decrypt content
    val encryptedContentKey: String,

    // Token that certifies the validity of this DRM context
    val token: String
)

File: r2-lcp-kotlin/src/main/cpp/lcp/DRMContext.cpp
#include "DRMContext.h"

namespace lcp
{
    DRMContext::DRMContext(
        const std::string & hashedPassphrase,
        const std::string & encryptedContentKey,
        const std::string & token
    ) {
        this->hashedPassphrase = hashedPassphrase;
        this->encryptedContentKey = encryptedContentKey;
        this->token = token;
    }
}
